### ISSUE TITLE
Change permalink structure and retain social proof count? 

### ISSUE THREAD

**Issue by byalextran** [copyblogger/genesis-simple-share#47]
I'd like to change the permalink structure of my site, but doing so would cause me to lose all share counts within the plugin. 

Two questions.

1) If and/or when you work on #27, is it possible to incorporate old permalink structures as an alternate URL for the total count? So the plugin would tally counts for the short link, current permalink and any previously used permalink structures. 

I know WordPress tracks old permalink structures because it automatically handles redirects for you. So I believe the data exists in WordPress to easily know what old URLs to check in the share count.  

2) If that's not doable, is there any custom code/hook I could utilize to manually add my old permalink structure to the share counts? 

Thanks!

**Comment by NicktheGeek** [comment]
We may have an option in the future to specify the share URL but I'm not sure that we will ever make it possible to define multiple URLs to check for previous shares. That adds a lot of overhead because you are not only making multiple calls to multiple services to get share counts, but you also have to make some method of dynamically updating the additional URLs. This ends up opening up a very slippery slope that will take a very light and fast plugin to make it cumbersome and bulky.

**Comment by byalextran** [comment]
totally understand keeping the plugin light/fast. 

i ended up finding a solution that solves my problem. posting it here in case anyone else finds it helpful. 

[here's a gist with the code](https://gist.github.com/byalextran/6758ab819b0d9083a650). 

put it at the bottom of your theme's `function.php` file and modify the date accordingly (and also the old permalink structure if it's not `/YYYY/mm/dd/`). 

in short, i'm hooking into the `get_permalink` function *only* when it's called within the plugin's `get_icon_output` function (where the permalink gets outputted). if the post was made before the permalink structure was changed, the permalink is returned with the old structure.

original code came from [this post by per soderlind](http://soderlind.no/wordpress-hook-into-another-hook/).
